src/platform/ESP32/ConnectivityManagerImpl_WiFi.cpp

```diff
@@ -402,8 +402,10 @@ CHIP_ERROR ConnectivityManagerImpl::InitWiFi()
             // Set a default station configuration.
             wifi_config_t wifiConfig;
             memset(&wifiConfig, 0, sizeof(wifiConfig));
-            memcpy(wifiConfig.sta.ssid, CONFIG_DEFAULT_WIFI_SSID, strlen(CONFIG_DEFAULT_WIFI_SSID) + 1);
-            memcpy(wifiConfig.sta.password, CONFIG_DEFAULT_WIFI_PASSWORD, strlen(CONFIG_DEFAULT_WIFI_PASSWORD) + 1);
+            memcpy(wifiConfig.sta.ssid, CONFIG_DEFAULT_WIFI_SSID,
+                   std::min(sizeof(wifiConfig.sta.ssid), strlen(CONFIG_DEFAULT_WIFI_SSID)));
+            memcpy(wifiConfig.sta.password, CONFIG_DEFAULT_WIFI_PASSWORD,
+                   std::min(sizeof(wifiConfig.sta.password), strlen(CONFIG_DEFAULT_WIFI_PASSWORD)));
```

Comment:
But ... now this is failing to copy the null-terminator, while `wifiConfig.sta.ssid` and `wifiConfig.sta.password` are supposed to be null-terminated.  @jadhavrohit924

Reply:
Yes, If we want to add null-terminator we can have ssid < 32 and pass < 64 bytes. But it can be equal to 32 and 64 bytes. We talked with wifi team, if ssid < 32 then null-terminator is being added and for ssid = 32 bytes it they are not adding null-terminator, same for pass.